packages/zoe/src/zoeService/zoe.js

```diff
@@ -71,18 +76,24 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
     getBrands: instance => instanceToInstanceAdmin.get(instance).getBrands(),
     getIssuers: instance => instanceToInstanceAdmin.get(instance).getIssuers(),
     getTerms: instance => instanceToInstanceAdmin.get(instance).getTerms(),
-    getInstance: invitation =>
-      E(invitationKit.issuer)
+    getInstance: async invitation => {
+      await assertInvitation(invitation);
+      return E(invitationKit.issuer)
         .getAmountOf(invitation)
-        .then(amount => amount.value[0].instance),
-    getInstallation: invitation =>
-      E(invitationKit.issuer)
+        .then(amount => amount.value[0].instance);
+    },
```

Comment:
This is asking the issuer twice.  I think one round trip is enough
```js
    getInstance: async invitation => {
      return E(invitationKit.issuer)
        .getAmountOf(invitation)
        .then(
          amount => amount.value[0].instance,
          () =>
            assert(
              false,
              details`A Zoe invitation is required, not ${invitation}`,
            ),
        );
    },
```
The helper function I extracted looks like this. You may prefer something else.
```js
  async function getAmountOfInvitationThen(invitation, onFulfill) {
    return E(invitationKit.issuer)
      .getAmountOf(invitation)
      .then(onFulfill, () =>
        assert(false, details`A Zoe invitation is required, not ${invitation}`),
      );
  }
```
If you feel that the cost of two round trips isn't as important as simple code (your version is simpler to read), I'm okay with that.

Reply:
It turns out this is better than the `getInterfaceOf` solution, since that requires an extra turn. I'll switch to this. Thanks for the code!